packages/fast-component-explorer/app/explorer.tsx

```diff
@@ -437,6 +545,36 @@ class Explorer extends Foundation<ExplorerHandledProps, {}, ExplorerState> {
     private get backgrounds(): typeof DarkModeBackgrounds | typeof LightModeBackgrounds {
         return DarkModeBackgrounds;
     }
+
+    private getNeutralPalette(colorSource: string): string[] {
+        const color: ColorRGBA64 | null = parseColor(colorSource);
+        if (color !== null) {
+            const hslColor: ColorHSL = rgbToHSL(color);
+            const augmentedHSLColor: ColorHSL | null = ColorHSL.fromObject({
+                h: hslColor.h,
+                s: hslColor.s,
+                l: 0.5,
+            });
+
+            if (augmentedHSLColor !== null) {
+                return createColorPalette(hslToRGB(augmentedHSLColor));
+            }
+        }
+        return DesignSystemDefaults.neutralPalette;
+    }
+
+    private handleUpdateTheme = (): void => {
+        this.setState({
+            theme:
+                this.state.theme === ThemeName.light ? ThemeName.dark : ThemeName.light,
+            viewConfig: merge({}, this.state.viewConfig, {
+                backgroundColor: this.state.theme === ThemeName.light ? dark : light,
+                neutralPalette: this.getNeutralPalette(
+                    this.state.theme === ThemeName.light ? dark : light
+                ),
```

Comment:
you seem to be doing `this.state.theme === ThemeName.light ? dark : light` over and over, can you declare a const above this that is `const updatedTheme: ThemeName = this.state.theme === ThemeName.light ? dark : light` and use updatedTheme?